tests/unit/test_utils.go

```diff
@@ -83,27 +94,41 @@ func (ls MockLedgerService) QueryDIDDoc(did string, version string) (*didTypes.D
 // TODO: add unit tests for testing QueryAllDidDocVersionsMetadata method.
 func (ls MockLedgerService) QueryAllDidDocVersionsMetadata(did string) ([]*didTypes.Metadata, *types.IdentityError) {
 	if ls.Did.Id == did {
-		return []*didTypes.Metadata{ls.Metadata}, nil
+		return ls.Metadata, nil
 	}
 
 	return nil, types.NewNotFoundError(did, types.JSON, nil, true)
 }
 
 func (ls MockLedgerService) QueryResource(did string, resourceId string) (*resourceTypes.ResourceWithMetadata, *types.IdentityError) {
-	if ls.Did.Id != did || ls.Resource.Metadata == nil || ls.Resource.Metadata.Id != resourceId {
+	if ls.Did.Id != did {
 		return nil, types.NewNotFoundError(did, types.JSON, nil, true)
 	}
 
-	return ls.Resource, nil
+	for _, resource := range ls.Resources {
+		if resource.Metadata == nil {
+			return nil, types.NewNotFoundError(did, types.JSON, nil, true)
+		}
+		if resource.Metadata.Id == resourceId {
+			return &resource, nil
+		}
+	}
+
+	return &resourceTypes.ResourceWithMetadata{}, types.NewNotFoundError(did, types.JSON, nil, true)
 }
 
 // TODO: add unit tests for testing QueryCollectionResources method.
 func (ls MockLedgerService) QueryCollectionResources(did string) ([]*resourceTypes.Metadata, *types.IdentityError) {
-	if ls.Did.Id != did || ls.Resource.Metadata == nil {
+	if ls.Did.Id != did {
 		return []*resourceTypes.Metadata{}, types.NewNotFoundError(did, types.JSON, nil, true)
 	}
 
-	return []*resourceTypes.Metadata{ls.Resource.Metadata}, nil
+	var metadata_list = make([]*resourceTypes.Metadata, len(ls.Resources))
```

Comment:
Rename this local variable from `metadata_list` to `metadataList` to match golang code style.